src/LanguageServer/Protocol/ExternalAccess/Razor/RazorComponentInfoHandler.cs

```diff
@@ -0,0 +1,133 @@
+    private readonly IGlobalOptionService _globalOptions;
+
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public RazorComponentInfoHandler(IGlobalOptionService globalOptions)
+    {
+        _globalOptions = globalOptions;
+    }
+
+    public bool MutatesSolutionState => false;
+
+    public bool RequiresLSPSolution => true;
+
+    public async Task<RazorComponentInfo?> HandleRequestAsync(RazorComponentInfoParams request, RequestContext context, CancellationToken cancellationToken)
+    {
+        var project = context.Solution?.GetProject(request.Project);
+
+        if (project is null)
+        {
+            return null;
+        }
+
+        var document = context.Solution.GetDocument(request.Document);
+        var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+        var syntaxTree = semanticModel.SyntaxTree;
+        var root = (CompilationUnitSyntax)syntaxTree.GetRoot(cancellationToken);
+
+        var (methods, fields) = GetInfoInsideRazorDocument(root, semanticModel, cancellationToken);
+
+        var result = new RazorComponentInfo
+        {
+            Methods = methods,
+            Fields = fields
+        };
+
+        return result;
+    }
+
+    private static (List<MethodInsideRazorElementInfo>, List<SymbolInsideRazorElementInfo>) GetInfoInsideRazorDocument(CompilationUnitSyntax root, SemanticModel semanticModel, CancellationToken cancellationToken)
+    {
+        var invocationExpressions = root.DescendantNodes().OfType<InvocationExpressionSyntax>().ToList();
+        var identifierNames = root.DescendantNodes().OfType<IdentifierNameSyntax>().ToList();
+        
+        List<MethodInsideRazorElementInfo> methods = [];
+        List<SymbolInsideRazorElementInfo> fields = [];
+        
+        foreach (var invocation in invocationExpressions)
+        {
+            var invocationOperation = semanticModel.GetOperation(invocation, cancellationToken) as IInvocationOperation;
+            var invocationDataFlow = semanticModel.AnalyzeDataFlow(invocation);
+            if (invocationOperation is null)
+            {
+                continue;
+            }
+
+            var targetMethod = invocationOperation.TargetMethod;
+            if (targetMethod is null)
+            {
+                continue;
+            }
+
+            var operationReturnType = invocationOperation.Type;
+            if (operationReturnType is null)
+            {
+                continue;
+            }
+
+            var parameterTypes = targetMethod.GetParameters().Select(parameter => parameter.Type.ToDisplayString()).ToList();
+            if (parameterTypes is null)
+            {
+                continue;
+            }
+
+            var methodInfo = new MethodInsideRazorElementInfo
+            {
+                Name = targetMethod.Name,
+                ReturnType = operationReturnType.ToNameDisplayString(),
```

Comment:
What happens with this string? I don't think `ToNameDisplayString()`, or `ToDisplayString()` used above, guarantee that the result will be valid C# code.